app/src/main/java/com/example/perfectweatherallyear/MainActivity.kt

```diff
@@ -19,21 +30,61 @@ class MainActivity : AppCompatActivity() {
 
         firebaseAnalytics = Firebase.analytics
 
-//        binding.crashButton.setOnClickListener {
-//            firebaseAnalytics.logEvent(FirebaseAnalytics.Event.SELECT_ITEM) {
-//                param(FirebaseAnalytics.Param.ITEM_ID, binding.crashButton.id.toString())
-//                param(FirebaseAnalytics.Param.ITEM_NAME, "CrashButton")
-//            }
-//            Firebase.crashlytics.log("this is my runtime exception")
-//            Firebase.crashlytics.setCustomKeys {
-//                key("my_log", "this is my runtime exception")
-//            }
-            //throw RuntimeException("Test Crash") // Force a crash
-//        }
+        getDynamicLinkFromFirebase()
+        openLinkFromNotification()
+    }
+
+    private fun openLinkFromNotification() {
+        val link = intent.getStringExtra("link")
+        if (link != null) {
+            val uri = Uri.parse(link)
+            getParamAndOpenDeepLink(uri)
+        }
+    }
+
+    private fun getDynamicLinkFromFirebase() {
+        Firebase.dynamicLinks.getDynamicLink(intent)
+            .addOnSuccessListener(this) { pendingDynamicLinkData: PendingDynamicLinkData? ->
+                var deepLink: Uri? = null
+
+                if (pendingDynamicLinkData != null) {
+                    deepLink = pendingDynamicLinkData.link
+
+                    if (deepLink != null) {
+                        getParamAndOpenDeepLink(deepLink)
```

Comment:
Can be simplified:
`pendingDynamicLinkData?.link?.run(::getParamAndOpenDeepLink)`